src/schema/validator.js

```diff
@@ -525,23 +530,152 @@ export class SchemaValidator {
     }
 
     validator.addKeyword({
-      keyword: 'max_manifest_version',
+      keyword: SCHEMA_KEYWORDS.MAX_MANIFEST_VERSION,
       // function of type SchemaValidateFunction (see ajv typescript signatures).
       validate: createManifestVersionValidateFn(
-        'max_manifest_version',
+        SCHEMA_KEYWORDS.MAX_MANIFEST_VERSION,
         (maxMV, manifestVersion) => maxMV >= manifestVersion
       ),
       errors: true,
     });
 
     validator.addKeyword({
-      keyword: 'min_manifest_version',
+      keyword: SCHEMA_KEYWORDS.MIN_MANIFEST_VERSION,
       validate: createManifestVersionValidateFn(
-        'min_manifest_version',
+        SCHEMA_KEYWORDS.MIN_MANIFEST_VERSION,
         (minMV, manifestVersion) => minMV <= manifestVersion
       ),
       errors: true,
     });
+
+    const checkForPrivilegedPermissionsValidate = (
+      keywordSchemaValue,
+      propValue
+    ) => {
+      const privilegedPermissions = this.getPrivilegedPermissionsSet(validator);
+      const found = new Set();
+
+      for (const permission of propValue) {
+        if (privilegedPermissions.has(permission)) {
+          found.add(permission);
+        }
+      }
+
+      const hasMozillaAddonsPermission = found.has('mozillaAddons');
+
+      // If the addon is expected to be privileged, we report a linting error if:
+      // - there are no privileged permissions required
+      // - and/or if the "mozillaAddons" permission isn't requested (which is going
+      //   to be a mandatory requirement even if there are also other privileged
+      //   permissions already required).
+      if (this.privileged) {
+        if (found.size === 0 || !hasMozillaAddonsPermission) {
+          checkForPrivilegedPermissionsValidate.errors = [
+            {
+              keyword: SCHEMA_KEYWORDS.VALIDATE_PRIVILEGED_PERMISSIONS,
+              params: {
+                postprocess: keywordSchemaValue,
+                privilegedPermissions: Array.from(found),
+                isPrivilegedAddon: this.privileged,
```

Comment:
I preferred to leave this to isPrivilegedAddon because there is also a privilegedPermissions and so I think that the current name is more immediately clear ("clear that it refers to the <<addon>> being privileged") then if it would be called more generically`isPrivileged`.

Given that it is an internal implementation detail it doesn't sound like a big deal that it doesn't match exactly how the cli option is named.

Reply:
yep, no problem.